### DISCUSSION TITLE
Quarkus with Kotlin: Use of runBlocking in @Produces

### DISCUSSION THREAD

**Discussion by TheBroach** [quarkusio/quarkus#43189] [Q&A] [answered]
Hello!
My question is essentially about CDI, but maybe it's interesting to see my real-world use case.

I am building a Quarkus application with Kotlin, and now I am working with some metrics stuff. The application is running in Amazon ECS, and I need the application to query the internal ECS Metadata Endpoint during the setup of Micrometer to be able to enrich all metrics with an "instance id". Since that call is asynchronous, and - from what I've understood? - CDI itself is not, I was thinking about simply doing this:
```
    @Produces
    @Singleton
    fun configureAllRegistries(): MeterFilter {
        val instanceId = runBlocking {
            instanceIdProvider.getInstanceId() ?: "no-instance-id"
        }
    // Use the value in the returned MeterFilter object
    }
```
The call to the endpoint should go extremely fast since it's (in some way) "internal".
Would this work? Or rather, is this a bad idea for some reason? :)
Thanks!

[a comment by an automated account is not shown]

**Comment by geoand** [comment, marked as answer]
It should work AFAICT

**Reply by TheBroach** [reply]
Thank you for the quick reply! Nice that it seems to be an OK solution. Follow-up question: Even if this particular call should be extremely fast, if I were to have another call that might take a bit more time, do I have some kind of "deadline" for how long I am allowed to block during the CDI process?

**Reply by geoand** [reply]
YW!

Nope, there is no deadline, it will just slow down application startup
